### ISSUE TITLE
`homeless-try` false positive on named error-union type aliases (default severity: error)

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#365]
# `homeless-try` false positive on named error-union type aliases (default severity: error)

- **Severity:** High (rule is on by default at `.err` — false positive fails builds)
- **Status:** Verified by reproduction (`zig ast-check` accepts the code; zlint reports an error)
- **Location:** `src/linter/rules/homeless_try.zig:140-143`

## The bug

The rule decides whether a function can contain `try` by syntactic inspection of the return type:

```zig
// homeless_try.zig:140
if (a.hasErrorUnion(ast, return_type)) return;
if (a.getRightmostIdentifier(ctx, return_type)) |ident| {
    if (std.mem.endsWith(u8, ident, "Error")) return;
}
```

`hasErrorUnion` only recognizes a literal `!` / `error_union` AST node, and the fallback heuristic only accepts identifiers ending in the string `"Error"`. A named alias for an error union satisfies neither:

```zig
const E = error{A};
const Result = E!void;      // alias, doesn't end in "Error"
fn bar() E!void { return error.A; }
pub fn foo() Result {       // valid Zig
    try bar();              // flagged: "`try` cannot be used in functions that do not return errors."
}
```

## User impact

The rule's category is `compiler` with `.default = .err`, so this false positive **fails lint runs out of the box** (exit 1) on perfectly valid, idiomatic code. Named result-type aliases are common in parsers, decoders, and public APIs (`pub const Result = ParseError!Node;`). Affected users must rename their alias to end in `Error` (wrong — it's a result type, not an error set) or disable a default-on rule.

## Reproduction

```sh
cat > ht.zig <<'EOF'
const E = error{A};
const Result = E!void;
fn bar() E!void {
    return error.A;
}
pub fn foo() Result {
    try bar();
}
EOF
zig ast-check ht.zig   # passes
zlint ht.zig           # error: homeless-try ... exit 1
```

## Proposed fix

When the return type is a plain identifier, resolve it with the semantic analyzer instead of string matching: `semantic.resolveBinding` the identifier, and if it resolves to a variable declaration whose initializer contains an error union (or error-set-returning expression such as `A!B`, `error{...}`, or another alias — one level of chasing covers the common case), treat the function as fallible. As a fallback, when the identifier resolves to something the analyzer can't classify (e.g. imported from another file or a comptime function result), the rule should **not** report — preferring false negatives over build-failing false positives for a `compiler`-category rule. The existing `endsWith("Error")` heuristic can remain as a fast path.
